services/authorization_helper.rb

```diff
@@ -0,0 +1,15 @@
+module AuthorizationHelper
+  def valid_open_key?(key)
+    return false if key.nil?
+
+    key_digest = Digest::MD5.hexdigest key
+    !Key.find_by(value: key_digest).nil?
+  end
```

Comment:
There is a method called `.present?` that could possible replace `!.nil?`, but they aren't exactly the same thing, so I am fine with `!.nil?` here.